src/main/java/kr/pickple/back/crew/service/CrewMemberService.java

```diff
@@ -68,6 +69,32 @@ public CrewProfileResponse findAllApplyForCrewMemberShip(final Long crewId, fina
         return CrewProfileResponse.fromEntity(crew, crewMemberRelationDtoList);
     }
 
+    @Transactional
+    public void permitCrewMemberShip(final Long crewId, final Long memberId, final CrewMemberPermitRequest crewMemberPermitRequest) {
+        final Crew crew = findByExistCrew(crewId);
+        final Member member = findMemberById(memberId);
+        //TODO:추후, 크루장인지 검증 로직 추가(11월 2일, 소재훈)
+
+        validateCrewMembership(member, crew, WAITING);
+
+        final CrewMember crewMember = crewMemberRepository.findByMemberAndCrew(member, crew).get();
+        crewMember.permitStatus(RegistrationStatus.from(crewMemberPermitRequest.getStatus()));
+    }
+
+    private void validateCrewMembership(final Member member, final Crew crew, final RegistrationStatus registrationStatus) {
+        final Optional<CrewMember> validateCrewMember = crewMemberRepository.findByMemberAndCrew(member, crew);
+
+        if (!validateCrewMember.isPresent()) {
+            throw new CrewException(CREW_MEMBER_STATUS_NOT_FOUND, member.getId());
+        }
+
+        final CrewMember crewMember = validateCrewMember.get();
+
+        if (crewMember.getStatus() != registrationStatus) {
+            throw new CrewException(CREW_MEMBER_ALREADY_EXISTED, member.getId());
+        }
+    }
```

Comment:
```suggestion
    @Transactional
    public void permitCrewMemberShip(final Long crewId, final Long memberId, final CrewMemberPermitRequest crewMemberPermitRequest) {
        final CrewMember crewMember = crewMemberRepository.findByMemberIdAndCrewId(memberId, crewId)
                .orElseThrow(() -> new CrewException(CREW_MEMBER_NOT_FOUND, memberId, crewId));
        
        crewMember.permitStatus(RegistrationStatus.from(crewMemberPermitRequest.getStatus()));
    }
```

- 이렇게 줄일 수 있을 것 같다는 생각이 드는데 혹시 어떠신가요?
- 어차피 crewId, memberId의 경우에는 해당 크루ID와 멤버ID를 통해 crewMember 객체를 가져오는 역할만 합니다.
- 그러면 각각을 findById를 해서 crew, member를 가져오면 벌써 2번의 불필요한 select 쿼리가 나가게 될 것 같습니다.
- 차라리 findByMemberIdAndCrewId 라는 메소드를 인터페이스에 직접 지정해서 crewId, memberId로 바로 crewMember 객체를 조회하는 게 어떨까 싶습니다.
- 그리고 여기에서는 CREW_MEMBER_STATUS_NOT_FOUND라는 예외 네이밍은 조금 어색한 것 같아요! crewId, memberId를 통해 crewMember 객체를 찾는 과정에서 없다는 예외이기 때문에 CREW_MEMBER_NOT_FOUND는 어떨까합니다~

Reply:
네! 해당 사항 반영하여 수정하였습니다~
반영 사항: 955367c42db620cc5be600668da4a38a69ba39dc